packages/compiler-cli/src/ngtsc/testing/src/utils.ts

```diff
@@ -138,3 +138,8 @@ function bindingNameEquals(node: ts.BindingName, name: string): boolean {
   }
   return false;
 }
+
+export function getSourceCodeForDiagnostic(diag: ts.Diagnostic): string {
+  const text = diag.file!.text;
+  return text.substr(diag.start!, diag.length!);
```

Comment:
```suggestion
  if (diag.file === undefined || diag.start === undefined || diag.length === undefined) {
    return '';
  }
  const text = diag.file.text;
  return text.substr(diag.start, diag.length);
```
Though this is pre-existing, we should avoid null assertions where possible. This particular helper function has no knowledge of whether these are defined, even if the larger context does. Or maybe it's desirable for this function to throw an error if those are `undefined`, in which case maybe we should still have the explicit check with a `throw new Error()`

Reply:
I think it's best to avoid the null assertion.